### DISCUSSION TITLE
Is mason-registry considered part of the "public API?"

### DISCUSSION THREAD

**Discussion by thetic** [mason-org/mason.nvim#168] [Q&A] [answered]
As I interpret [the docs](https://github.com/williamboman/mason.nvim/blob/main/doc/reference.md), the answer is no, but both [mason-lspconfig](https://github.com/williamboman/mason-lspconfig.nvim/blob/84aff881ee525288297a381c6e5998ffddaa43ed/lua/mason-lspconfig/init.lua#L28=) and [mason-tool-installer](https://github.com/WhoIsSethDaniel/mason-tool-installer.nvim/blob/main/lua/mason-tool-installer/init.lua#L1=) rely on it to retrieve package specs by name. Is this a stable interface? Is there an alternative approach that users and plugins should use instead?

**Comment by WhoIsSethDaniel** [comment]
The docs, specifically the image in this doc, https://github.com/williamboman/mason.nvim/blob/main/doc/reference.md, show several registry methods that are public .

**Comment by williamboman** [comment, marked as answer]
Hello! Yep it is! The public functions are available in `:h mason`
